modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/management_toolbar/LinkOrButton.js

```diff
@@ -7,54 +7,119 @@ import ClayButton from '@clayui/button';
 import ClayIcon from '@clayui/icon';
 import ClayLink from '@clayui/link';
 import classNames from 'classnames';
-import React from 'react';
-
-const LinkOrButton = ({
-	ariaLabel,
-	children,
-	className,
-	disabled,
-	href,
-	symbol,
-	title,
-	wide,
-	...otherProps
-}) => {
-	const responsive = symbol && children;
-	const Wrapper = href && !disabled ? ClayLink : ClayButton;
+import React, {useEffect, useMemo, useState} from 'react';
 
-	return (
-		<>
-			<Wrapper
-				aria-label={symbol && ariaLabel}
-				block={otherProps.button?.block}
-				className={classNames(className, {
-					'd-md-none': responsive,
-					'nav-btn-monospaced': responsive,
-					'pl-4 pr-4': wide && !symbol,
-				})}
-				disabled={disabled}
-				href={href}
-				{...otherProps}
-				title={symbol && title}
+import './LinkOrButton.scss';
+
+const DEFAULT_BREAKPOINT_MD = '768px';
+
+/**
+ * A dynamic link/button that includes a symbol, a label, or both of them. All
+ * props are optional except for the label prop.
+ *
+ * It renders a ClayButton or a ClayLink depending on the provided:
+ * - disabled state (a ClayButton will be used).
+ * - href (a ClayLink will be used).
+ * - otherwise a ClayButton will be used.
+ *
+ * Then it shows a symbol on smaller screens, and label on larger screens.
+ * When only the symbol is being shown, an aria-label and title are added to the
+ * button for accessibility purposes. If there is no provided title, the label
+ * prop will be reused as title.
+ *
+ * It will always add a title to the button if it is specified as prop.
+ */
+function LinkOrButton(
+	{
+		className = '',
+		disabled = false,
+		href = '',
+		label,
+		symbol = '',
+		title = '',
+		...otherProps
+	},
+	ref
+) {
+	const [element, setElement] = useState(null);
+	const TagName = href && !disabled ? ClayLink : ClayButton;
+
+	const breakpointMd = element
+		? window.getComputedStyle(element).getPropertyValue('--breakpoint-md')
+		: DEFAULT_BREAKPOINT_MD;
```

Comment:
Not this part, but the useMediaQuery hook does. The nice part is that it only triggers when the media query matches (or when it stops matching) once. That way we don't render the component hundreds of times during a resize.

Moreover React does not care if the component is being rerender because of a resize or anything else, we can rely on CSS to put any complex condition (I find this kinda cool 🤓)

Reply:
TIL how `window.matchMedia` works. Nice!